### DISCUSSION TITLE
predict_cumulative_hazard_function gives same result for every sample (CoxPHSurvivalAnalysis))

### DISCUSSION THREAD

**Discussion by drewlehe** [sebp/scikit-survival#228] [Q&A]
I follow the documentation for the CoxPH model exactly, my dataset is a DataFrame with only floats. After calling .fit() and predict_cumulative_hazard_function() every row has the same cumulative hazard, but it does not do that for the wha500 example.
Each sample has a different survival curve.
It would be useful to have a method for calling the predicted hazard at each timestep.

**Comment by sebp** [comment]
Could you please provide a minimal working example so I can reproduce 
the issue?

**Reply by drewlehe** [reply]
```from sklearn.model_selection import train_test_split
from sksurv.linear_model import CoxPHSurvivalAnalysis
X_train, X_test, y_train,  y_test = train_test_split(data, datay, random_state=40)
cox = CoxPHSurvivalAnalysis()
cox = cox.fit(X_train, y_train)
cox.predict_cumulative_hazard_function(X_test)```

It's really simple, `data`  is a pandas DataFrame and `datay` is a numpy structured array. The only unusual thing I can think of is that, in this example, there are no censored observations.

**Comment by drewlehe** [comment]
I see now how you pulled the cumulative hazard at each timestep, and I implemented some code to pull the hazard at each timestep fromthe cumulative hazard. But I'm getting hazards over 100% for some timesteps. 
I can see examples where the cumulative hazard grows by more than 1, in one timestep, so there's a hazard over 100% (it's always on the last timestep for that observation).
```
test_hazards = cox.predict_cumulative_hazard_function(X_test)
cumulative_hazards = []
for fn in test_hazards:
    cumulative_hazards.append(fn(fn.x).tolist())
```

**Reply by sebp** [reply]
The cumulative hazard function does _not_ return probabilities, but _rates_, therefore any non-zero value, including larger 1, is perfectly fine.

**Reply by drewlehe** [reply]
I see. So to get h(t), take the difference of the differences from t to t+1.
Can we add a function specifically for hazard values? I tried calculating the PMF and dividing f(t) by S(t), but also got values over 1 in some cases.

**Reply by sebp** [reply]
To get the hazard function, you would need to compute the derivative of the cumulative hazard function, or use `h(t) = f(t)/S(t)`, as you mentioned. As I mentioned before, the hazard function is not a probability density function, therefore values above 1 are fine.
